tests/bluemira/magnetostatics/test_circuits.py

```diff
@@ -138,6 +149,110 @@ def test_mixedsourcesolver():
+    pd_inputs = {"x1": {"value": 4}, "x2": {"value": 16}, "dz": {"value": 0}}
+
+    pf_inputs = {
+        "x1": {"value": 5},
+        "x2": {"value": 10},
+        "z1": {"value": 10},
+        "z2": {"value": -9},
+        "ri": {"value": 0.4},
+        "ro": {"value": 1},
+    }
+    ta_inputs = {
+        "x1": {"value": 4},
+        "dz": {"value": 0},
+        "sl": {"value": 6.5},
+        "f1": {"value": 3},
+        "f2": {"value": 4},
+        "a1": {"value": 20},
+        "a2": {"value": 40},
+    }
+
+    wtf_inputs = {
+        "x_mid": {"value": 0.4},
+        "x_curve_start": {"value": 2.5},
+        "x_out": {"value": 9.5},
+        "z_mid_up": {"value": 7.5},
+        "z_mid_down": {"value": -7.5},
+        "z_max_up": {"value": 11},
+        "z_max_down": {"value": -11},
+    }
+
+    parameterisations = [
+        PrincetonD,
+        TripleArc,
+        PictureFrame,
+        FullDomeFlatInnerCurvedPictureFrame,
+    ]
+    p_inputs = [pd_inputs, ta_inputs, pf_inputs, wtf_inputs]
+    clockwises = [False] * len(p_inputs) + [True] * len(p_inputs)
+    p_inputs = p_inputs * 2
+    parameterisations = parameterisations * 2
+
+    @pytest.mark.parametrize(
+        "parameterisation, inputs, clockwise",
+        zip(parameterisations, p_inputs, clockwises),
+    )
+    def test_geometry_ccw(self, parameterisation, inputs, clockwise):
+        shape = parameterisation(inputs).create_shape()
+        coords = shape.discretize(ndiscr=50, byedges=True)
+        coords.set_ccw((0, -1, 0))
+        if clockwise:
+            coords.set_ccw((0, 1, 0))
+        circuit = ArbitraryPlanarRectangularXSCircuit(
+            coords,
+            0.25,
+            0.5,
+            1.0,
+        )
+        open_circuit = ArbitraryPlanarRectangularXSCircuit(
+            coords[:, :25].T, 0.25, 0.5, 1.0
+        )
+        n_chain = int(self._calc_daisychain(circuit))
+        n_sources = len(circuit.sources) - 1
+        assert n_chain == n_sources
+        assert self._check_continuity(circuit.sources[-1], circuit.sources[0])
+        assert self._calc_daisychain(open_circuit) == len(open_circuit.sources) - 1
+
+    def test_circle(self):
+        shape = make_circle(5, (0, 9, 0), axis=(0, 0, 1))
+        coords = shape.discretize(ndiscr=30, byedges=True)
+        coords.set_ccw((0, 0, -1))
+        circuit = ArbitraryPlanarRectangularXSCircuit(
+            coords,
+            0.25,
+            0.5,
+            1.0,
+        )
+        assert self._calc_daisychain(circuit) == len(circuit.sources) - 1
+        assert self._check_continuity(circuit.sources[-1], circuit.sources[0])
+        coords.set_ccw((0, 0, 1))
+        circuit = ArbitraryPlanarRectangularXSCircuit(
+            coords,
+            0.25,
+            0.5,
+            1.0,
+        )
+        assert self._calc_daisychain(circuit) == len(circuit.sources) - 1
+        assert self._check_continuity(circuit.sources[-1], circuit.sources[0])
```

Comment:
Could these be two separate tests? Perhaps it could be parametrized as you've done above?

Also, as a general point on test naming, I'd try and be more specific about what behaviour the test is actually testing. What about `circle` are we testing? Maybe something like `test_a_circuit_from_a_clockwise_circle_is_continuous`? I find that reading `test_` as `test_that_` in my head helps me construct better names.